### ISSUE TITLE
A question about training

### ISSUE THREAD

**Issue by isty2e** [leela-zero/leela-zero#1190]
During my halted 128x20 training run, I did some experiments with SWA. In the middle of training, a plain SGD network and a SWA network were sampled, and training was restarted with those networks respectively.

In general, SWA networks had less L2 loss, and for them the total loss tends to increase at the early stage, so convergence was a bit slower than SGD counterparts, with L2 loss increasing back again.

With the most of recent promoted networks being SWA ones, and the networks are being trained from the last best network, I wonder if restarting training from the corresponding SGD network can be helpful in terms of strength. Supposing the learning rate is not too high, this can lead to a faster convergence, but training from a SWA network might be less vulnerable under overfitting.

**Comment by ThorAvaTahr** [comment]
Hmm jeah, I wonder anyway what we are doing here. We seem to have followed an empirical approach (hé it works let's do it) (please don't take me wrong, that's fine!), but we don't, or at least I don't understand why it works.  Apparently there is some kind of mechanism that drives SGD away from an optimum of strength, but applying SWA brings it back again. I could imagine this trick would work like a couple of times, helping SGD evolve quickly more quickly towards an optimum, but here it seems that SWA is consistently stronger than SGD. 

I wonder whether FPU and other MCTS effects cause to narrow PV's and to narrow policies, when SGD'ing them, while the averaging smears them out again. I am not sure if we could measure this, perhaps we can. Anyway, I would like to understand this better.

**Comment by gcp** [comment]
> Apparently there is some kind of mechanism that drives SGD away from an optimum of strength, but applying SWA brings it back again. 

SGD isn't driven away from an optimum of strength, but particularly with high learning rates it will not reach the global optimum. SWA is a technique that can estimate it better by averaging. No magic here.

**Comment by sethtroisi** [comment]
closing, no active discussion for ~1year
